app/src/test/java/edu/wpi/first/shuffleboard/app/components/NetworkTableTreeTest.java

```diff
@@ -99,12 +97,11 @@ public void testBranches() {
   @Test
   public void testSort() {
     // deliberately not in order
-    table.putString("c", "");
-    table.putString("sub_a/sub_entry_a", "");
-    table.putString("a", "");
-    table.putString("b", "");
-    table.putString("sub_b/sub_entry_b", "");
-    waitForNtcoreEvents();
+    waitForNtcoreEvents(() -> table.putString("c", ""));
+    waitForNtcoreEvents(() -> table.putString("sub_a/sub_entry_a", ""));
+    waitForNtcoreEvents(() -> table.putString("a", ""));
+    waitForNtcoreEvents(() -> table.putString("b", ""));
+    waitForNtcoreEvents(() -> table.putString("sub_b/sub_entry_b", ""));
```

Comment:
The future will complete when the listener for `table.putString("c", "");` is called but not for `table.putString("sub_b/sub_entry_b", "");`